connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1150,6 +1221,32 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
+  test("Retain empty recursive proto fields when retain.empty.message.types=true") {
+    // This verifies that a empty proto like 'message A { A a = 1}' can be retained by
+    // inserting a dummy field.
+
+    val emptyProtoSchema =
+      StructType(StructField("__dummy_field_in_empty_struct", StringType) :: Nil)
+    val expectedSchema = StructType(
+      StructField("empty_proto",
+        StructType(
+          StructField("recursive_field", emptyProtoSchema) ::
+            StructField("recursive_array", ArrayType(emptyProtoSchema, containsNull = false)
+          ) :: Nil
+        )
+      ) :: Nil
+    )
+
+    val options = Map("recursive.fields.max.depth" -> "2", "retain.empty.message.types" -> "true")
```

Comment:
Will the schema change if max depth is higher, say 3?

Reply:
yes, schema will change for recursive empty field if the option changed.

Reply:
Could you change it to 3 or add a case with 3? I just want to see how the schema changes. Please add comment to make it easier to see the schema.